snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/version/SnomedVersioningRequest.java

```diff
@@ -145,60 +146,75 @@ protected void doVersionComponents(TransactionContext context) throws Exception
 	
 	private void versionComponents(TransactionContext context, Iterable<? extends SnomedDocument> componentsToVersion, Multimap<String, String> componentIdsByReferringModule) {
 		
-		for (SnomedDocument componentToVersion : componentsToVersion) {
-			// register IDs for publication
-			if (componentToVersion instanceof SnomedComponentDocument) {
-				componentIdsToPublish.add(componentToVersion.getId());
-			}
+		try (TransactionContext ctx = new RepositoryTransactionContext(context, context.author(), "Versioning components", context.parentLock())) {
```

Comment:
Did `context.openTransaction` not work? I'd prefer using it instead of instantiating a concrete context implementation.